### ISSUE TITLE
`.find(undefined)` returns the first record

### ISSUE THREAD

**Issue by bingtsingw** [romeerez/orchid-orm#76]
`find(undefined)` should throw an error, but it return a the first record. Even worse, `find(undefined).update(dto)` now updates all records without any query filter.

**Comment by romeerez** [comment]
`find(undefined)` is a TypeScript error, isn't it?

If TS doesn't catch it in your case, then it's a bug and I'll fix it, but let me know the context.

And if you suppress and trick the TS, it will lead to errors in various places, not only `find`, but any function in principle would work incorrectly if you bypass the TS check and pass a wrong value.

**Comment by bingtsingw** [comment]
Well, I thought it was a small bug, but I didn't expect it to be an interesting and in-depth topic. I am very interested in discussing it.

First of all, is my use case, I get the id from user request pseudocode `const id = request('id')`, this is `any` type, so I assert it to `string` like this `const id = request('id') as string`, as you can guess, this can be `undefined` sometimes.
This was definitely my mistake, as I should validate user input with something like `zod`, but I forgot it.

So,  let's further discuss if we should move the validation part to the library.

Basically I think it's a design trade-off and there is no right answer, and I totally agree with you not to add all checks across the whole codebase, but I'm wondering if we could add some checks for critical parts, such as the `find(undefined).update(dto)` case is too dangerous that may alter all records.

**Comment by romeerez** [comment]
> Such as the find(undefined).update(dto) case is too dangerous that may alter all records.

I agree, the `find` should have a check for null/undefined inside.

**Comment by romeerez** [comment]
Published, now the `find` will throw if you pass a `null` or `undefined`.
